### ISSUE TITLE
Ktlint warnings on generated Apollo code

### ISSUE THREAD

**Issue by tasomaniac** [apollographql/apollo-android#2079]
We have Ktlint setup using: https://github.com/JLLeitschuh/ktlint-gradle

We have other modules with code generation using `kapt`. Ktlint does not detect those.

When we switched to Apollo Kotlin generation, it looks like `generated` folder is detected. I am guessing it is because the generated code is added as a regular source directory (see below) and not marked as generated.

https://github.com/apollographql/apollo-android/blob/master/apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/JvmTaskConfigurator.kt#L67

**Comment by GurpreetSK95** [comment]
Is someone working on this?

**Comment by tasomaniac** [comment]
Not really. I'm not sure what to do about this. I'm suspecting it is something to do with generated code being marked as regular sources but not generated.

**Comment by bryanstern** [comment]
I'm running into this as well. A field in my schema has been deprecated, but despite utilizing the following plugin options kotlinc will warn on it as well, which means I need to disable `-Werror` for the whole project. I think other projects have worked around this by suppressing all warnings on their generated code.

```
warnOnDeprecatedUsages
failOnWarnings
```

**Comment by BoD** [comment]
Closing as the generated code shouldn't have warnings per #5242 available in [v4.0.0-beta.1](https://github.com/apollographql/apollo-kotlin/releases/tag/v4.0.0-beta.1)
